text/0000-uisize.md

```diff
@@ -0,0 +1,128 @@
+- Feature Name: `uptr_iptr`
+- Start Date: 2016-06-02
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+`isize` and `usize` either must be a `size_t` like type, or `intptr_t` like
+type. Most uses of `usize` and `isize` are as `size_t` or `ptrdiff_t`
+like-types (`Vec<T>`, `[T]`, `.offset`, etc.). Therefore, remove the guarantee
+that `usize` and `isize` are pointer-size, and make them exclusively
+object-sized.
+
+# Motivation
+[motivation]: #motivation
+
+We want to support the embedded space, and new CPUs like the CHERI. These CPUs
+do not support `usize` == `uptr`, and, in the case of the CHERI, don't support
+`uptr` at all. Most CPUs don't actually support the idea that `uptr == usize`:
+just the currently popular ones.
+
+We really want to support usecases where `uptr != usize`. The embedded space,
+where segmented architectures are still very popular, is a very good case for
```

Comment:
@comex 

> These days I hear there's rarely all that much advantage (in power or cost) to using 8-bit MCUs instead of a 32-bit ARM Cortex-M0 for any applications, let alone ones involving programs big enough for the ability to use cross-platform code to matter.

Sorry, can you give me the summary of this. :) i.e., when I read your comment, my impression is that you are saying that, in the embedded space, most things are moving to use more familiar 32-bit architectures, is that right?